src/OaiPmh/Harvester.php

```diff
@@ -248,7 +272,9 @@ public function launch()
 
         // If we made it this far, all was successful. Save last harvest info
         // and clean up the stored state.
-        $this->stateManager->saveDate($explicitHarvestEndDate);
+        if (!isset($this->stopAfter) || !$this->stopAfter >= 0) {
```

Comment:
What is the purpose of this condition? Is the idea that we don't want to save a date when we're only doing a partial harvest? If so, I wonder if a slightly more concise way of phrasing it would be:

`if (($this->stopAfter ?? -1) >= 0) {`

Might also be worth expanding the comment above to note something like "(unless we have a limit imposed by stopAfter)".